crypto/commitments/damgard-fujisaki.go

```diff
@@ -133,13 +133,14 @@ func NewDamgardFujisakiReceiver(safePrimeBitLength, k int) (*DamgardFujisakiRece
 // NewDamgardFujisakiReceiverFromExisting returns an instance of receiver with the same
 // parameters as the receiver used as an input. Different instances are needed because
 // each sets its own Commitment value.
-func NewDamgardFujisakiReceiverFromExisting(receiver *DamgardFujisakiReceiver) (
+func NewDamgardFujisakiReceiverFromParams(qrSpecialRSA *groups.QRSpecialRSA, H, G *big.Int,
```

Comment:
May I ask why this change was introduced? I mean, why are we now specifying every parameter instead of just copying the receiver's values? Test code is much more verbose now.

Reply:
Changed to (h, g, k).

Because of this call:
receiver2, err := commitments.NewDamgardFujisakiReceiverFromParams(receiver.QRSpecialRSA, receiver.H, c1, receiver.K)

Note that there is c1 used instead of receiver.G. But also - receiver might not always be available.